### PR TITLE
A small prototype for the Zeeguu reader app

### PR DESCRIPTION
**Pull request by JorritO**

### CODE REVIEW

File: Zeeguu Reader/Zeeguu Reader/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(0.1 * Double(NSEC_PER_SEC)))
+		dispatch_after(delayTime, dispatch_get_main_queue()) { () -> Void in
+			if (!ZeeguuAPI.sharedAPI().isLoggedIn) {
+				self.presentLogin()
+			}
+		}
+		return true
+	}
+	
+	func setupArticleRootViewcontroller() {
```
Comment (mircealungu) [inline Zeeguu Reader/Zeeguu Reader/AppDelegate.swift]: camelcase Controller...

File: Zeeguu Reader/Zeeguu Reader/ArticleListViewController.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		
+		self.tableView.estimatedRowHeight = 80
+		
+		self.title = "APP_TITLE".localized
+		self.navigationItem.title = "APP_TITLE".localized
+		
+		let addButton = UIBarButtonItem(barButtonSystemItem: .Add, target: self, action: "insertNewObject:")
+		self.navigationItem.rightBarButtonItem = addButton
+		
+		let logoutButton = UIBarButtonItem(title: "Logout", style: .Done, target: self, action: "logout:")
```
Comment (mircealungu) [inline Zeeguu Reader/Zeeguu Reader/ArticleListViewController.swift]: "LOGOUT".localized?

File: Zeeguu Reader/Zeeguu Reader/ArticleView.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+	var article: Article!
+	
+	convenience init(article: Article) {
+		self.init()
+		self.article = article;
+		self.translatesAutoresizingMaskIntoConstraints = false
+		
+		
+
```
Comment (mircealungu) [inline Zeeguu Reader/Zeeguu Reader/ArticleView.swift:41]: a bit too much white space? :)

File: Zeeguu Reader/Zeeguu Reader/ArticleView.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+			}
+		}
+		
+		
+		let views = ["title":titleLabel, "content": contentLabel]
+		
+		self.addSubview(titleLabel)
+		self.addSubview(contentLabel)
+		
+		self.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat("H:|-[title]-|", options: NSLayoutFormatOptions(rawValue: 0), metrics: nil, views: views))
```
Comment (mircealungu) [inline Zeeguu Reader/Zeeguu Reader/ArticleView.swift:87]: The constraint language is very interesting!

File: Zeeguu Reader/Zeeguu Reader/ZGTextView.swift [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		self.article = article;
+		self.translatesAutoresizingMaskIntoConstraints = false
+	}
+	
+	func selectedText() -> String {
+//		return self.stringByEvaluatingJavaScriptFromString("window.getSelection().toString()")!
+		return self.textInRange(self.selectedTextRange!)!
+	}
+	
+	func selectedTextContext() -> String {
```
Comment (mircealungu) [inline Zeeguu Reader/Zeeguu Reader/ZGTextView.swift:45]: this looks cool.
I could think of a few interesting test cases for you if you want :) but then you'd have to make this function testable... which means, take selected range and text as arguments.

File: Zeeguu Reader/Zeeguu Reader/Localizable.strings [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
+"APP_TITLE" = "Zeeguu";
```
Comment (mircealungu) [inline Zeeguu Reader/Zeeguu Reader/Localizable.strings]: "Zeeguu Reader"

File: Zeeguu Reader/Zeeguu Reader/Localizable.strings [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+"BASE_LANGUAGE" = "Base language";
+"BOOKMARK" = "Bookmark";
+"CHOOSE_LANGUAGE" = "Choose language";
+"EMAIL" = "Email";
+"LEARN_LANGUAGE" = "Learn";
+"LOGIN" = "Login";
```
Comment (mircealungu) [inline Zeeguu Reader/Zeeguu Reader/Localizable.strings:33]: Missing LOGOUT

File: Zeeguu Reader/Zeeguu Reader/AppDelegate.swift (hunk 1 above)
Comment (JorritO) [inline Zeeguu Reader/Zeeguu Reader/AppDelegate.swift]: Fixed

File: Zeeguu Reader/Zeeguu Reader/ArticleListViewController.swift (hunk 2 above)
Comment (JorritO) [inline Zeeguu Reader/Zeeguu Reader/ArticleListViewController.swift]: The button was just there to make it quick to logout, but as the button will be somewhere in the future, I fixed it

File: Zeeguu Reader/Zeeguu Reader/ArticleView.swift (hunk 3 above)
Comment (JorritO) [inline Zeeguu Reader/Zeeguu Reader/ArticleView.swift:41]: Added a file that will make Github show the tabs as 4 spaces

File: Zeeguu Reader/Zeeguu Reader/ArticleView.swift (hunk 4 above)
Comment (JorritO) [inline Zeeguu Reader/Zeeguu Reader/ArticleView.swift:87]: Also look at this if you want to know a bit more about it:

https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/AutolayoutPG/VisualFormatLanguage.html

File: Zeeguu Reader/Zeeguu Reader/ZGTextView.swift (hunk 5 above)
Comment (JorritO) [inline Zeeguu Reader/Zeeguu Reader/ZGTextView.swift:45]: Done

File: Zeeguu Reader/Zeeguu Reader/Localizable.strings (hunk 6 above)
Comment (JorritO) [inline Zeeguu Reader/Zeeguu Reader/Localizable.strings]: Done

File: Zeeguu Reader/Zeeguu Reader/Localizable.strings (hunk 7 above)
Comment (JorritO) [inline Zeeguu Reader/Zeeguu Reader/Localizable.strings:33]: Fixed, as explained above.

**Pull request merged**